x/mailbox/keeper/genesis.go

```diff
@@ -26,6 +26,8 @@ func (k Keeper) InitGenesis(ctx sdk.Context, gs types.GenesisState) error {
 	for _, msgDelivered := range gs.DeliveredMessages {
 		k.Delivered[msgDelivered.Id] = true
 	}
+
+	k.SetDomain(ctx, gs.Domain)
```

Comment:
How will the domain initially be set for existing chains?

Reply:
App initialization doesn't have a context. So would it need to be set in an upgrade handler?